src/client/components/homeComponents/HomePosts.jsx

```diff
@@ -0,0 +1,81 @@
+import { useState, useEffect } from "react";
+
+const URL = process.env.REACT_APP_API_URL;
+const postEndpoint = "/post";
+const postURL = URL + postEndpoint;
+
+const HomePosts = () => {
+  const [postArr, setPostArr] = useState([]);
+  const [search, setSearch] = useState("");
+
+  const handleSearch = e => {
+      setSearch(e.target.value);
+  }
+
+  const searchPosts = (searchTerm, postArr) => {
+    if (!searchTerm) {
+      return postArr;
+    }
+    return postArr.filter(post => post.content.toLowerCase().includes(searchTerm) || post.title.toLowerCase().includes(searchTerm) || post.profile.user.username.toLowerCase().includes(searchTerm))
+  }
+
+  const filteredPosts = searchPosts(search, postArr);
+
+  const changeDate = oldDate => {
+      const tIndex = oldDate.indexOf("T");
+      const dayMonthYear = oldDate.slice(0, tIndex);
+      const pointIndex = oldDate.indexOf(".");
+      const time = oldDate.slice(tIndex + 1, pointIndex)
+      const newDate = `${dayMonthYear}, ${time}`
+      return newDate;
+  }
```

Comment:
Dates are tricky to work with but generally they're easier if you pass the date string into a new `Date` object:

```js
const newDate = new Date(oldDateString);
return newDate.toDateString(); // will return "Mon Feb 28 2022"
```

There are a lot of methods on the `Date` object that make it easier to format strings instead of slicing and searching indexes: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date